qcodes/instrument_drivers/rohde_schwarz/HMC804x.py

```diff
@@ -0,0 +1,115 @@
+
+        select_cmd = ":INSTrument:NSELect {};".format(channel)
+
+        self.add_parameter("set_voltage",
+                           label='Target voltage output',
+                           set_cmd="{} :SOURce:VOLTage:LEVel:IMMediate:AMPLitude {}".format(
+                               select_cmd, '{}'),
+                           get_cmd="{} :SOURce:VOLTage:LEVel:IMMediate:AMPLitude?".format(
+                               select_cmd),
+                           get_parser=float,
+                           unit='V',
+                           vals=vals.Numbers(0, 32.050)
+                          )
+        self.add_parameter("set_current",
+                           label='Target current output',
+                           set_cmd="{} :SOURce:CURRent:LEVel:IMMediate:AMPLitude {}".format(
+                               select_cmd, '{}'),
+                           get_cmd="{} :SOURce:CURRent:LEVel:IMMediate:AMPLitude?".format(
+                               select_cmd),
+                           get_parser=float,
+                           unit='A',
+                           vals=vals.Numbers(0.5e-3, self._parent.max_current)
+                           )
+        self.add_parameter('state',
+                           label='Output enabled',
+                           set_cmd='{} :OUTPut:CHANnel:STATe {}'.format(select_cmd, '{}'),
+                           get_cmd='{} :OUTPut:CHANnel:STATe?'.format(select_cmd),
+                           val_mapping={'ON': 1, 'OFF': 0},
+                           vals=vals.Enum('ON', 'OFF')
+                           )
+        self.add_parameter("voltage",
+                           label='Measured voltage',
+                           get_cmd="{} :MEASure:SCALar:VOLTage:DC?".format(
+                               select_cmd),
+                           get_parser=float,
+                           unit='V',
+                          )
+        self.add_parameter("current",
+                           label='Measured current',
+                           get_cmd="{} :MEASure:SCALar:CURRent:DC?".format(
+                               select_cmd),
+                           get_parser=float,
+                           unit='A',
+                           )
+        self.add_parameter("power",
+                           label='Measured power',
+                           get_cmd="{} :MEASure:SCALar:POWer?".format(
+                               select_cmd),
+                           get_parser=float,
+                           unit='W',
+                           )
+
+
+class RohdeSchwarzHMC804x(VisaInstrument):
+    """
+    This is the qcodes driver for the Rohde & Schwarz HMC804x Power Supply
+    """
+
+    _max_currents = {3: 3.0, 2: 5.0, 1: 10.0}
+
+    def __init__(self, name, address, num_channels, **kwargs):
+        super().__init__(name, address, **kwargs)
+
+        self.max_current = RohdeSchwarzHMC804x._max_currents[num_channels]
```

Comment:
should this be self rather than RohdeSchwarzHMC804x?

Reply:
Is a static member of the class, not a variable of the object, so the reference to RohdeSchwarzHMC804x is correct